tests/Unit/Framework/ActionTesting.hpp

```diff
@@ -518,37 +607,47 @@ class MockCollectionOfDistributedObjectsProxy {
 
   template <typename Action, typename... Args>
   void threaded_action(std::tuple<Args...> args) noexcept {
-    if (mock_distributed_objects_->size() != 1) {
-      ERROR("NodeGroups must have exactly one element during testing, but have "
-            << mock_distributed_objects_->size());
-    }
-    mock_distributed_objects_->begin()->second.template threaded_action<Action>(
-        std::move(args));
+    static_assert(std::is_same_v<ChareType, MockNodeGroupChare>,
+                  "Do not call threaded_action for other than a Nodegroup");
+    alg::for_each(*mock_distributed_objects_,
+                  [&args](auto& index_and_mock_distributed_object) noexcept {
+                    index_and_mock_distributed_object.second
+                        .template threaded_action<Action>(args);
+                  });
```

Comment:
Because this is a member function of MockCollectionOfDistributedObjectsProxy, i.e. a proxy to the entire chare, not to one element of the chare.  So just like simple_action, shouldn't threaded_action be a broadcast call to all elements?  If you want one element, you would use ckLocalBranch to get your local element (or index the MockCollectionOfDistributedObjectsProxy to get a different element if you want that), and that would return a MockDistributedObject (in the case of ckLocalBranch) or a proxy to one (in the case of indexing) that you could call threaded_action on for a single element.

Pls let me know if my understanding of threaded_action is incorrect.

Reply:
You can't run a threaded action on an entire component, and you can't even run it on a remote element.  I thought you couldn't even run it on a proxy, but our documentation and code disagree with each other on this:

docs/DevGuide/Parallelization.md:
> User controlled threading is possible by calling the non-entry method member function `threaded_action`.

src/Parallel/Algorithms/AlgorithmNodegroup.ci:
> template \<typename Action>
> entry void threaded_action();

@nilsdeppe which of these is correct?

Reply:
@wthrowe If we aren't supposed to ever run a threaded action on an entire component, then `MockCollectionOfDistributedObjectsProxy::threaded_action` should just be removed; so that's what I did in a fixup commit.

Reply:
The way we implement threaded entry methods they can be run on an entire component. I don't think we use that feature right now. I see no reason right now to disable it either. The threaded actions should behave (almost) entirely the same way as simple actions except that they don't lock the entire group.